### PR TITLE
BFD-1848: Stop writing the access.log File

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-1848](https://jira.cms.gov/browse/BFD-1848)

**User Story or Bug Summary:**

Stop writing to the access.log to save disk space but continue writing the access.json

---

### What Does This PR Do?

Remove writer to access.log and replace with unstructured log with suppressed output.

### What Should Reviewers Watch For?

Does this code successfully not write an access.log file and only to an access.json file and meet the functional requirements.

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by lsmitchell — COMMENTED** [review]
Just a minor doc fix. Nice to knock out some tech debt here!

File: apps/bfd-server/bfd-server-launcher/src/main/java/gov/cms/bfd/server/launcher/DataServerLauncherApp.java [hunk 1]
```diff
@@ -301,31 +302,26 @@ public static void main(String[] args) {
...
-   * TODO: BFD-1844 Remove access.log.
    */
   private static class BfdRequestLog extends CustomRequestLog {
     /**
      * Construct a BFD Request Log.
      *
-     * @param accessLogFileName filename for the unstructured log 'access.log'
-     * @param accessLogFormat format for the unstructured log 'access.log'
+     * @param writer source for the unstructured log
+     * @param accessLogFormat format for the unstructured log
```
[the remark this answers is not in the document]
Comment (JeremyIR) [reply apps/bfd-server/bfd-server-launcher/src/main/java/gov/cms/bfd/server/launcher/DataServerLauncherApp.java]: Updated

**Comment by JeremyIR** [comment]
I have now added the feature to clear the request log object just like we do the MDC in a way that doesnt take up too much memory.

Additionally, after further inspecting the access.json for the launcher, we are testing for all the mdc fields present (There are only 2).

As far as access.json for the bfd-server-war, there is no way to test it properly because of the unpredictability of what gets logged each line to mdc e.g mdc_bene_id. So Trotter had no interest in pursuing an IT for validating access.json within the server-war directory because of this ability to not parse properly in real time.

**Comment by lsmitchell** [comment]
> As far as access.json for the bfd-server-war, there is no way to test it properly because of the unpredictability of what gets logged each line to mdc e.g mdc_bene_id. So Trotter had no interest in pursuing an IT for validating access.json within the server-war directory because of this ability to not parse properly in real time.

can you explain further about the unpredictability? If we ran a query and waited 5-10 seconds, would we not have the information logged we needed (even if it happened to be out of order or something?)

**Pull request closed without merging**